src/otg/dataset/dataset.py

```diff
@@ -80,11 +80,11 @@ def validate_schema(self: Dataset) -> None:
             )
 
         # Required fields not in dataset
-        required_fields = [
-            (x.name, x.dataType) for x in expected_schema if not x.nullable
-        ]
+        required_fields = [x.name for x in expected_schema if not x.nullable]
         if missing_required_fields := [
-            x for x in required_fields if x not in observed_fields
+            req
+            for req in required_fields
+            if not any(field.name == req for field in observed_fields)
```

Comment:
Wouldn't something like this work and clean the sourcery issue?
`if any(field.name != req ...)`

Reply:
No. Saying that `any(field.name != req ...)` will always* evaluate to True, because in this second loop you are iterating through the observed_fields.
Say that `required_fields = ["A", "B"]` and `observed_fields = ["B", "C"]`:

- When req = "A", the second loop will be `[True, True]`, so the any() condition will be True, which is correct because "A" is indeed missing. 
- When req = "B", the second loop will be `[False, True]`, so `any()` will be True and therefore "B" will count as missing, when it is not true.

By writing `not any()`, we are really evaluating that `all` in required are observed_fields. (But I can't use `all` in the second loop because I'm looping through the observed_fields). 

It is a bit of a tongue twister, but it is the best I got. The evaluation could be simple (`all(any(field.name == req for field in observed_fields) for req in required_fields)`), but the fact that we want to list which fields are missing makes the logic more complex.